spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -220,7 +221,8 @@ private void writeAndValidateWithLocations(Table table, File location, File expe
     return spark.internalCreateDataFrame(JavaRDD.toRDD(rdd), convert(schema), false);
   }
 
-  @Test
+  // This fails due to SPARK-28730
+  @Ignore
```

Comment:
This test fails because the requirement is enforced by Spark, not by Iceberg. I'm adding an `Assume` so the test is still valid for Spark 2.x but won't be run for 3.0.